lib/pbench/test/functional/agent/cli/commands/tools/test_tools_list.py

```diff
@@ -259,6 +259,26 @@ def tools_on_multiple_hosts(self, pbench_run):
                     tool = p / "perf"
                     tool.write_text("--record-opts='-a --freq=100'")
 
+    # Issue #3434
+    @pytest.fixture
+    def labels_on_multiple_hosts(self, pbench_run):
+        group = "default"
+
+        host = "th1.example.com"
+        label = pbench_run / f"tools-v1-{group}" / host / "__label__"
+        label.write_text("foo")
+
+        host = "th2.example.com"
+        label = pbench_run / f"tools-v1-{group}" / host / "__label__"
+        label.write_text("bar")
+
+        group = "test"
+        host = "th1.example.com"
+        label = pbench_run / f"tools-v1-{group}" / host / "__label__"
+        label.write_text("bar")
+
+        # th2 has no label
```

Comment:
The spacing here isn't consistent enough to convey the structure.  I think it would be better to have a couple of comments and/or a docstring.  (E.g., line 280 doesn't say "in group test"; and, it's not immediately clear that what you're doing here is setting up a matrix with `group` on one axis and `host` on the other.)